.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,68 @@
+# **what?**
+# Checks that a file has been committed under the /.changes directory
+# as a new CHANGELOG entry.  Cannot check for a specific filename as
+# it is dynamically generated by change type and timestamp.
+# This workflow should not require any secrets since it runs for PRs
+# from forked repos.
+# By default, secrets are not passed to workflows running from
+# a forked repo.
+
+# **why?**
+# Ensure code change gets reflected in the CHANGELOG.
+
+# **when?**
+# This will run for all PRs going into main and *.latest.
+
+name: Check Changelog Entry
+
+on:
+  push:
+    branches:
+      - "main"
+      - "*.latest"
+  pull_request:
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  changelog:
+    name: changelog
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository (PR)
```

Comment:
This step will never be executed b/c you are triggering on a `pull_request` event and not a `pull_request_target`. It looks like this step is always skipped in the test runs you have done. The next step for the path filters doesn't need to repo to be checked out and checks the PR directly it seems so we can remove this all together I'm guessing